refactor: open new PR

njsmithfm opened the pull request.

njsmithfm commented:
**TASKS**

- [x] Refine the title and the contextual paragraph
- [x] Scatterplot Map
- [ ] Remove all the deaths
- [x] fix tooltips
- [x] Color the dots by borough in the same way as the histogram chart
- [x] Remove the Dropdown
- [x] Update the chart titles accordingly
- [ ] refilter the API for 30 days from current date
- [ ] General Design Tweaks
- [x] left 1/3 column for text, right 2/3 columns for charts (stacked)

njsmithfm commented:
@Kully do you know how to customize plotly express tooltips? 

I'm hoping to get as close as I can back to this:
![image](https://github.com/user-attachments/assets/a8d97932-42a8-4f64-8a74-44c14d135b53)

here's what I've got currently
![image](https://github.com/user-attachments/assets/d7d81f86-23bb-467a-b62c-c623c6a0d2d6)

Here's the plotly code right now:
```
def create_map_fig(df, days):
    map_fig = px.scatter_map(
        df,
        lat="latitude",
        lon="longitude",
        color="borough",
        hover_name=pd.to_datetime(df["crash_date"]),
        hover_data={
            "number_of_cyclist_injured",
            "vehicle_type_code1",
            "vehicle_type_code2",
            "contributing_factor_vehicle_1",
        },
        labels={"number_of_cyclist_injured": "Cyclists Injured", 
                "vehicle_type_code1": "Vehicle 1",
                "vehicle_type_code2":"Vehicle 2",
                "contributing_factor_vehicle_1": "Contributing Factor"},
        center=dict(lat=40.7128, lon=-73.9560),
        zoom=10,
        map_style="dark",
        title=f"Cyclist Injury Locations (Last {days} Days)",
    )

    return map_fig

njsmithfm commented:
solved some of the tooltip formatting:

![image](https://github.com/user-attachments/assets/072d0778-bce9-4f7d-9297-2fa572d6a70c)

However I'm still confused how to format the date correctly from within the px.scatter_map function. It's formatted earlier in the app with this:
`df["crash_date"] = pd.to_datetime(df["crash_date"])`

but within the scattermap function it's still showing as the default timestamp. 

```
def create_map_fig(df, days):
    map_fig = px.scatter_map(
        df,
        lat="latitude",
        lon="longitude",
        color="borough",
        'crash_date'= pd.to_datetime(df["crash_date"]),
        hover_data={
            "crash_date": True,
            "latitude": False,
            "longitude": False,
            "number_of_cyclist_injured": True,
            "vehicle_type_code1": True,
            "vehicle_type_code2": True,
            "contributing_factor_vehicle_1": True,
        },
        labels={
            "crash_date": 'Date',
            "borough": 'Borough',
            "number_of_cyclist_injured": "Cyclists Injured",
            "vehicle_type_code1": "Vehicle 1",
            "vehicle_type_code2": "Vehicle 2",
            "contributing_factor_vehicle_1": "Contributing Factor",
        },
        center=dict(lat=40.7128, lon=-73.9560),
        zoom=10,
        map_style="dark",
        title=f"Cyclist Injury Locations (Last {days} Days)",
    )
    
    return map_fig

```

Kully commented:
> @Kully do you know how to customize plotly express tooltips?

Look at the `labels` parameter https://plotly.github.io/plotly.py-docs/generated/plotly.express.scatter_map.html

If you want to customize something like, using colon `:` rather than equals sign `=`, I would look into the `hovertemplate` property in Plotly.

Kully commented:
> solved some of the tooltip formatting:

Ahh I see you actually found the `labels` param, nice stuff!

> However I'm still confused how to format the date correctly from within the px.scatter_map function. It's formatted earlier in the app with this: `df["crash_date"] = pd.to_datetime(df["crash_date"])`
> 
> but within the scattermap function it's still showing as the default timestamp.

Hmm, I see what you are saying. I think you have to look at D3-Formatting. Check it out in the docs.

Kully commented:
@njsmithfm Can you rename this PR to something more informative?

<img width="1274" alt="image" src="https://github.com/user-attachments/assets/b2f84c7e-a93f-4492-9904-1cba05c0c28c" />

njsmithfm commented:
> @njsmithfm Can you rename this PR to something more informative?
> <img alt="image" width="1274" src="https://private-user-images.githubusercontent.com/10369095/436715558-b2f84c7e-a93f-4492-9904-1cba05c0c28c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU0MzkyODcsIm5iZiI6MTc0NTQzODk4NywicGF0aCI6Ii8xMDM2OTA5NS80MzY3MTU1NTgtYjJmODRjN2UtYTkzZi00NDkyLTk5MDQtMWNiYTA1YzBjMjhjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDIzVDIwMDk0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg1YzZkYzE2YTc1YTNjNGE0YTA5NDUzMmZmZGUwODZiZmJjYTNjODUwNjcxZGQyYTViOGMyNzdkMThmMmIyYjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Bax4DcwpfKcBYZvt3EU_hcUc0mYyoUW37mb-F-5L8oQ">

yeah--I think there was a mismatch between naming it from my terminal and when I opened it up in github

njsmithfm commented:
@Kully I am going to close this pull request and bump one/two tasks for the next one--fixing the API filtering to limit to most recent 30 days and to separate the deaths from the injuries. Let me know if you have anything to add!

To summarize this PR I cleaned the dataset, implemented the scattermap, and got a consistent borough colorscheme across both graphs.

Kully commented:
> @Kully I am going to close this pull request and bump one/two tasks for the next one--fixing the API filtering to limit to most recent 30 days and to separate the deaths from the injuries. Let me know if you have anything to add!
> 
> To summarize this PR I cleaned the dataset, implemented the scattermap, and got a consistent borough colorscheme across both graphs.

Looking now.

Kully commented on constants.py:
```diff
@@ -4,3 +4,33 @@
+        "contributing_factor_vehicle_1": "Contributing_Factor",
+    }
+)
+NYC_BIKE_API_LINK = NYC_BIKE_API_LINK[
```
Good work on adding the `constants.py` file! 👍

Kully commented on main.py:
```diff
@@ -16,64 +16,68 @@
 
-df =  constants.NYC_BIKE_API_LINK
+days = 30
+df = constants.NYC_BIKE_API_LINK
```
Happy you are using the constants, but I would import the variables you need from the constants. It looks cleaner.

```python
from .constants import (
    NYC_BIKE_API_LINK,
    ...
)
```

Kully commented on main.py:
```diff
@@ -16,64 +16,68 @@
-import constants
 
-df =  constants.NYC_BIKE_API_LINK
+days = 30
```
Should days be in the constants file? If we are resurrecting the dropdown later, it could make sense to leave here.

Kully reviewed (changes requested):
This is great! I checked out the app and it feels so much better! 👏 I like the dark mode - feels very modern.

I have a few ideas for a next iteration after using it:

1. Is the 'marked with an "X"' blurb in the paragraph true? I don't know what this is referring to? If old, I would cut.

<img width="1235" alt="image" src="https://github.com/user-attachments/assets/1963516a-526f-4748-a98d-1ee6e5233b7a" />

2. I found myself flipping up and down to compare what's in the map, and what's in the histogram.

I know we agreed on stacking them for now, but now after using it, I am thinking more like a user, and am wishing they were both next to each other.

Perhaps an actionable step is:
- when the screen size is small, stack them (i.e. mobile)
- when above small screen size, make each graph width=6 using bootstrap component (i.e. tablet and desktop)

3. I like how the histogram is binned by week, and also like how the map is showing the last 30 days. However, there is a bit of a visual disconnect now, since there isn't a 1 to 1 on the data in map and the data in histogram.

This is okay, but just expressing there is a bit of a cognitive gap here that the user has to take on.

I would start thinking about how to address this, and find a way to tighten the relationship between them (we can get on a call for this ... a big one)

For merging, just had a couple of requested changes. 🙏
